src/js/actions/libraries.js

```diff
@@ -499,6 +497,145 @@ define(function (require, exports) {
+            tempPreviewPath;
+        
+        return Promise
+            .bind(this)
+            .then(function () {
+                var graphicEditStatus = this.flux.stores.library.getGraphicEditStatus(element);
+                
+                // If the element already have a temp file, update the temp file path.
+                if (graphicEditStatus) {
+                    tempFilePath = graphicEditStatus.documentPath;
+                    tempPreviewPath = graphicEditStatus.previewPath;
+                    return;
+                }
+                
+                // Otherwise, create a temp copy of the element for edit.
+                return _getTempPaths()
+                    .bind(this)
+                    .then(function (paths) {
+                        var tempFilename = this.flux.stores.library.generateTempFilename(element);
+                        
+                        tempFilePath = paths.tempBasePath + pathUtil.sep + tempFilename;
+                        tempPreviewPath = tempFilePath.replace(".psd", "p.png");
+                    })
+                    .then(function () {
+                        return Promise.fromNode(function (done) {
+                            element.getPrimaryRepresentation().getContentPath(done);
+                        }).then(function (contentPath) {
+                            if (!contentPath) {
+                                return Promise.reject("Failed to fetch content of element: " + element.name);
+                            }
+                            
+                            return contentPath;
+                        });
+                    })
+                    .then(function (contentPath) {
+                        return this.flux.actions.export.copyFile(contentPath, tempFilePath);
+                    });
+            })
+            .then(function () {
+                var previewSetting = {
+                    path: tempPreviewPath,
+                    width: RENDITION_GRAPHIC_SIZE,
+                    height: RENDITION_GRAPHIC_SIZE
+                };
+                
+                // Open the graphic asset's temp file and tell PS to generate an updated 
+                // preview whenever the file is saved
+                return this.flux.actions.documents.open(tempFilePath, {
+                    externalPreview: previewSetting
+                });
+            })
+            .then(function () {
+                var documentID = this.flux.stores.application.getCurrentDocumentID(),
+                    payload = {
+                        documentID: documentID,
+                        documentPath: tempFilePath,
+                        previewPath: tempPreviewPath,
+                        element: element
+                    };
+
+                return this.dispatchAsync(events.libraries.OPEN_GRAPHIC_FOR_EDIT, payload);
+            })
+            .catch(function (e) {
+                log.debug("[CC Lib] " + e);
```

Comment:
Maybe a log.warn instead, and a better `[header]` letting us know it's the openGraphicForEdit function.